lib/ts/recipe/thirdparty/api/implementation.ts

```diff
@@ -182,42 +84,21 @@ export default function getAPIInterface(): APIInterface {
                 createdNewUser: response.createdNewUser,
                 user: response.user,
                 session,
-                authCodeResponse: accessTokenAPIResponse.data,
+                oAuthTokens: oAuthTokensToUse,
+                rawUserInfoFromProvider: userInfo.rawUserInfoFromProvider,
             };
         },
 
-        appleRedirectHandlerPOST: async function ({ code, state, options }): Promise<void> {
-            const redirectURL =
-                options.appInfo.websiteDomain.getAsStringDangerous() +
-                options.appInfo.websiteBasePath.getAsStringDangerous() +
-                "/callback/apple?state=" +
-                state +
-                "&code=" +
-                code;
-            options.res.sendHTMLResponse(
-                `<html><head><script>window.location.replace("${redirectURL}");</script></head></html>`
-            );
+        appleRedirectHandlerPOST: async function ({ formPostInfoFromProvider, options }): Promise<void> {
+            const queryString = qs.stringify(formPostInfoFromProvider);
+            const state = formPostInfoFromProvider.state;
+            const stateObj = JSON.parse(state);
+            const redirectURI = stateObj.redirectURI;
+            const redirectURL = `${redirectURI}?${queryString}`;
+
+            options.res.setHeader("Location", redirectURL, false);
+            options.res.setStatusCode(303);
+            options.res.sendHTMLResponse("");
```

Comment:
double check if this is required at all and if the content-type being set in the response is correct or not.

Reply:
I tried a basic express app. Browser ignores the content-type and also the content, and redirect works as expected.